pkg/providers/imagefamily/nodeimage.go

```diff
@@ -0,0 +1,84 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagefamily
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter-provider-azure/pkg/operator/options"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+type NodeImage struct {
+	ID           string
+	Requirements scheduling.Requirements
+}
+
+type NodeImages []NodeImage
+
+type NodeImageProvider interface {
+	List(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass) (NodeImages, error)
+}
+
+func (p *Provider) List(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass) (NodeImages, error) {
+	supportedImages := getSupportedImages(nodeClass.Spec.ImageFamily)
+	if options.FromContext(ctx).UseSIG {
```

Comment:
Possibly a question for @tallaxes -- is this the preferred way to pass "parameters"?

I'm asking because I am not a huge fan - I would rather see `List(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass, options *SomeOptions)` (or some similar pattern) which at least makes it clear that this is an input to this function.

Reply:
I agree that I'd like that pattern more. This was simply following the pattern I saw for retrieving UseSIG from elsewhere.

This actually did cause a panic during my testing, since I missed this input wasn't set for one of my tests.

Reply:
Yes, we would strongly prefer to extract these parameters from context at the earliest reasonable opportunity. Lower-level functions ideally should not depend on where the setting is coming from.

Reply:
Added a TODO comment to address this.

I think it makes the most sense to be setting this for the provider at construction time. However, will do this as a refactor to constructor in a follow up PR since it touches some of the existing code paths too.